### PR TITLE
Add rate limiting and use 4 workers

### PR DESCRIPTION
**Pull request by sarayourfriend**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -7,3 +7,12 @@ services:
       - 5005:5000
     env_file:
       - .env
+    links:
+      - redis
+    depends_on:
+      - redis
```
Comment (AetherUnbound) [inline docker-compose.yml:13]: I usually tend to put the dependent service at the top

File: docker-compose.yml [hunk 2]
```diff
@@ -7,3 +7,12 @@ services:
       - 5005:5000
     env_file:
       - .env
+    links:
+      - redis
```
Comment (AetherUnbound) [inline docker-compose.yml:11]: I _think_ this line is redundant

File: tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import pytest
+from quart_rate_limiter.store import MemoryStore
+
+
+@pytest.fixture
+def app():
+    from app import limiter, app
+    limiter.store = MemoryStore()
```
Comment (sarayourfriend) [inline tests/conftest.py:9]: Ohhhh nice

**Review by sarayourfriend — COMMENTED** [review]
This makes sense to me, definitely easier than trying to get Redis working in the test environment?

But why wouldn't redis connect anyway when doing:

```
docker compose run --rm parler-search pytest
```

**Comment by AetherUnbound** [comment]
I've removed the local docker environment, since the tests should be able to run locally again now.

Instead of spinning up a redis instance, I've added an `app` fixture which represents the Quart app. The app is modified prior to tests to use a `MemoryStore` rather than anything involving redis.

**Comment by AetherUnbound** [comment]
> This makes sense to me, definitely easier than trying to get Redis working in the test environment?
> 
> But why wouldn't redis connect anyway when doing:
> 
> ```
> docker compose run --rm parler-search pytest
> ```

This.... should work :thinking: let me do some poking

**Comment by AetherUnbound** [comment]
Oh, we probably have to add the tests to the container and install the dev requirements in the container and I think it's probably just easier to mock redis :sweat:

**Review by AetherUnbound — APPROVED** [review]
Feels weird approving my own code :sweat_smile: but looks great!!

**Comment by sarayourfriend** [comment]
> Oh, we probably have to add the tests to the container and install the dev requirements in the container and I think it's probably just easier to mock redis 😓

I did do this but.. still didn't work 🤷 oh well, definitely easier like you said.

**Pull request merged**